File: 1512. Number of Good Pairs/1512. Number of Good Pairs.cpp
/*
 * Problem: 1512. Number of Good Pairs
 * Difficulty: Easy
 * Link: https://leetcode.com/problems/number-of-good-pairs/submissions/1984968239/
 * Language: cpp
 * Date: 2026-04-22
 */

class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int good_pairs = 0;
        for(int i = 0;i<nums.size();i++){
            for(int j = 0;j<nums.size();j++){
                if(i<j && nums[i]==nums[j]){
                    good_pairs++;
                }
            }
        }

        return good_pairs;
    }
};

File: 1512. Number of Good Pairs/1512. Number of Good Pairs.py
#
# Problem: 1512. Number of Good Pairs
# Difficulty: Easy
# Link: https://leetcode.com/problems/number-of-good-pairs/
# Language: python
# Date: 2026-04-22


class Solution(object):
    def numIdenticalPairs(self, nums):
        good_pairs = 0
        for i in range(0,len(nums)):
            for j in range(0,len(nums)):
                if (i<j and nums[i]==nums[j]):
                    good_pairs += 1

        return good_pairs
        
